pkg/graveler/retention/garbage_collection_manager.go

```diff
@@ -24,14 +24,17 @@ const (
 	commitsFileSuffixTemplate   = "%s/retention/gc/commits/run_id=%s/commits.csv"
 )
 
-type KVGarbageCollectionManager struct {
+type GarbageCollectionManager struct {
 	blockAdapter                block.Adapter
 	refManager                  graveler.RefManager
 	committedBlockStoragePrefix string
 	db                          db.Database
+	commitIteratorCreator       iteratorCreator
 }
 
-func (m *KVGarbageCollectionManager) GetCommitsCSVLocation(runID string, sn graveler.StorageNamespace) (string, error) {
+type iteratorCreator func(ctx context.Context, repoID graveler.RepositoryID) (graveler.CommitIterator, error)
```

Comment:
done

Reply:
Why not create the function in the ref manager level? We already pass it to the GC. This will require us to add 2 func parameters to the signature of the GC. One for commit iterator and one for branch